sbtc-core/src/signer/mod.rs

```diff
@@ -0,0 +1,116 @@
+/// An sBTC signer configuration
+pub mod config;
+
+use std::collections::HashMap;
+
+use crate::SBTCError;
+use crate::{signer::config::Config, SBTCResult};
+use bitcoin::{Network, PrivateKey, PublicKey, Transaction as BitcoinTransaction};
+use p256k1::ecdsa;
+use url::Url;
+use wsts::{bip340::SchnorrProof, common::Signature};
+
+#[derive(Default, Clone, Debug)]
+/// Signers' data required for weighted distributed signing
+pub struct Signers {
+    /// Signer id to public key mapping
+    pub signer_ids: HashMap<u32, ecdsa::PublicKey>,
+    /// Vote share to public key mapping
+    pub vote_shares: HashMap<u32, ecdsa::PublicKey>,
+}
+
+/// sBTC Signer trait
+pub trait SBTCSigner {
+    fn signers(&self) -> SBTCResult<Signers>;
+    fn coordinator_public_key(&self, _tx: BitcoinTransaction) -> SBTCResult<PublicKey>;
```

Comment:
Alternately, if you put the token `PublicKeys` into the trait name, then we could drop `_public_key` from the fn names.